kong/plugins/hmac-auth/access.lua

```diff
@@ -129,6 +134,22 @@ local function validate_clock_skew(headers, date_header_name, allowed_clock_skew
   return true
 end
 
+local function validate_request_body(headers, body)
+  local sha_recieved = headers[DIGEST]
+
+  if body == nil then
+    return true
+  elseif sha_recieved == nil then
+    return false
```

Comment:
@vaibhavatul47  here is what I am thinking to change it to 
```
-- If request body validation is enabled, request may require `digest` validation 
 local digest_recieved = headers[DIGEST]
  if conf.validate_request_body and digest_recieved then

    if not validate_digest(digest_recieved) then
      return false, {status = 403, message = SIGNATURE_NOT_VALID }
    end
  end


-----------------------
local function validate_digest(headers)

  local sha_recieved = headers[DIGEST]
  -- Client didn't send `digest` header
  -- as it is not interested in body validation
  if not sha_recieved then
    return true
  end

  req_read_body()
  local body = req_get_body_data()
  -- request must have body as client sent
  -- a digest header
  if not body then
    return false
  end

  local sha256 = resty_sha256:new()
  sha256:update(body)
  local sha_created = "SHA-256=" .. ngx_encode_base64(sha256:final())

  return sha_created == sha_recieved
end
```

I will also work on adding support for other hashing algorithm, at least SHA1 and SHA-256 for both Digest and HTTP signature validation